### PR TITLE
Split the next five Code Dojo roots

### PR DESCRIPTION
**Pull request by flyingrobots**
Lowers WF-165 Code Dojo debt from 52 to 42 by splitting the next five double-counted roots behind stable public facades. Preserves exports and behavior, adds focused regression and cycle evidence, and updates the living roadmap.

Validated by the full pre-push gate: 905 test files, 4,056 tests, workspace typechecks, strict Code Dojo checks, zero ESLint findings, and all scripted interactive examples.

Advances #480.

## Summary by CodeRabbit

* **New Features**
  * Added interactive DAG rendering with styled terminal layout, edge highlighting, accessibility text, and pipe-format output.
  * Enhanced the performance-gradient demo with modular rendering, deterministic noise presets, and memory stats.
  * Expanded the batteries-included TUI app shell with drawer overlays, themed skeleton components, and a scripted interaction/testing driver.
* **Bug Fixes**
  * Improved runtime crash handling and render scheduling stability; improved framebuffer reuse and initial resize behavior.
  * Fixed DAG rendering edge-case for empty badges and corrected detour-width calculation.
* **Documentation**
  * Updated Code Dojo milestone/burndown narratives and the unreleased changelog entries.
* **Tests**
  * Added/expanded coverage for DAG, runtime, skeleton overlays, perf charting, and Code Dojo tranche rules.

### CODE REVIEW

File: packages/bijou/src/core/components/dag-render-metrics.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    const fromLayer = layerMap.get(node.id);
+    const fromColumn = columnIndex.get(node.id);
+    if (fromLayer == null || fromColumn == null) continue;
+    for (const childId of node.edges ?? []) {
+      const toLayer = layerMap.get(childId);
+      const toColumn = columnIndex.get(childId);
+      if (
+        toLayer == null
+        || toColumn == null
+        || fromColumn === toColumn
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply packages/bijou/src/core/components/dag-render-metrics.ts]: Fixed in commit 84dcb247. A focused regression now proves same-column skip edges reserve detour width.

File: packages/bijou-tui/src/runtime-pipeline.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      state.ctx.io,
+      state.currentSurface,
+      state.targetSurface,
+      state.ctx.style,
+      state.outBuf,
+    );
+    next();
+  });
+  pipeline.use('Output', (_state, next) => {
+    buffers.swap();
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply packages/bijou-tui/src/runtime-pipeline.ts]: Fixed in commit 801a2d73. The Output stage now adopts the final middleware-selected target surface, with an integration regression.

File: examples/perf-gradient/perf-noise.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    if (attenuation <= 0) return 0;
+    const index =
+      permuted(permuted(xsb + offsetX) + ysb + offsetY) & 0x0e;
+    attenuation *= attenuation;
+    return attenuation * attenuation
+      * (gradient(index) * dx + gradient(index + 1) * dy);
+  };
+  for (let index = 0; index < 256; index++) source[index] = index;
+  let state = (seed * 1664525 + 1013904223) | 0;
+  for (let index = 0; index < 3; index++) {
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply examples/perf-gradient/perf-noise.ts]: Fixed in commit 61837a83. The pre-split LCG warm-up sequence and exact deterministic noise samples are restored and pinned.

File: packages/bijou/src/core/components/dag-render-text.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import type { DagNode } from './dag.js';
+import { buildLayerArrays } from './dag-layout.js';
+
+export function renderPipe(nodes: DagNode[]): string {
+  if (nodes.length === 0) return '';
+  const labels = new Map(nodes.map((node) => [node.id, node.label] as const));
+  const lines: string[] = [];
+  for (const node of nodes) {
+    const edges = node.edges ?? [];
+    const badge = node.badge == null ? '' : ` (${node.badge})`;
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply packages/bijou/src/core/components/dag-render-text.ts]: Fixed in commit d1deb1e2. Empty badges now lower exactly like absent badges in pipe, accessible, boxed, and compact modes.

File: packages/bijou-tui/src/driver-harness.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      bus.dispose();
+      throw error;
+    }
+  })();
+  const state: HarnessState<Model, M> = {
+    app,
+    options,
+    bus,
+    clock,
+    start: clock.now(),
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply packages/bijou-tui/src/driver-harness.ts]: Fixed in commit e2a981c0. Harness timing now starts before setup and app initialization, with deterministic clock evidence.

**Comment by flyingrobots** [comment]
## Review remediation complete

All 18 review findings are repaired in append-only commits, every inline thread
is resolved, and exact-head local plus pre-push verification is green.

| Source | Severity | Area | Commit | Outcome |
| --- | --- | --- | --- | --- |
| CodeRabbit | Actionable | Performance chart reference ownership and warm-up alignment | `ab062a64` | Fixed and regression-tested |
| CodeRabbit | Actionable | Skeleton footer separator | `c4287da5` | Fixed and snapshot-tested |
| CodeRabbit | Actionable | Narrow quit-modal width | `cc19acdf` | Clamped and regression-tested |
| CodeRabbit | Actionable | Secondary runtime crash output | `cb3e422c` | Suppressed and regression-tested |
| CodeRabbit | Actionable | Synchronous render-clock bookkeeping | `2f29606c` | Fixed and regression-tested |
| CodeRabbit | Actionable | Rejected shutdown drain | `9dfa6765` | Consumed and regression-tested |
| CodeRabbit | Actionable | Startup resize crash surface | `7b46b3e7` | Preserved and regression-tested |
| CodeRabbit | Nitpick | Runtime issue type import | `629b3bc9` | Hoisted to the contract import |
| CodeRabbit | Nitpick | Differ output allocation | `7aa5836c` | Capacity now reused |
| CodeRabbit | Nitpick | Shared performance-noise contract | `b90d3505` | One noise field and density ramp |
| CodeRabbit | Nitpick | Shared performance surface type | `ae2c1b60` | One named type across modules |
| CodeRabbit | Nitpick | Skeleton main-pane identity | `320313ce` | Centralized through one helper |
| CodeRabbit | Nitpick | Skeleton contract cycles | `07f923e4` | Broken through dependency-free leaves |
| Codex | P1 | Same-column DAG detour width | `84dcb247` | Restored and regression-tested |
| Codex | P2 | Middleware-replaced runtime target | `801a2d73` | Adopted and integration-tested |
| Codex | P2 | Noise permutation warm-up | `61837a83` | Pre-split field restored and pinned |
| Codex | P2 | Empty DAG badges | `d1deb1e2` | Omitted in every lower mode |
| Codex | P2 | Harness initialization timing | `e2a981c0` | Included and deterministically tested |

Supporting closeout:

- `282799cd` regenerates the two affected frame snapshot rows through the
  owning Vitest update path.
- `6d359b9d` records the complete review witness in the changelog and WF-165
  design record.
- Code Dojo debt: `42` = `27` file/context + `15` code-size + `0` mock +
  `0` ESLint.
- Exact-head suite: 914 test files, 4,077 tests, all passing.
- Pre-push firewall: DOGFOOD catalog/debt gates, size/type gates, full suite,
  and scripted interactive examples all passed.
- Review threads: 0 unresolved.

@codex review please

File: examples/perf-gradient/perf-noise.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      permuted(permuted(xsb + offsetX) + ysb + offsetY) & 0x0e;
+    attenuation *= attenuation;
+    return attenuation * attenuation
+      * (gradient(index) * dx + gradient(index + 1) * dy);
+  };
+  for (let index = 0; index < 256; index++) source[index] = index;
+  let state = (seed * 1664525 + 1013904223) | 0;
+  for (let index = 0; index < 2; index++) {
+    state = (state * 1664525 + 1013904223) | 0;
+  }
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply examples/perf-gradient/perf-noise.ts:35]: Not applied: this suggestion would restore the regression. At merge-base ed8c4c7b52f2b596312cfb079f3e0c456f19c5ba, examples/perf-gradient/main.ts lines 236-240 perform one initial seed assignment followed by exactly two warm-up transitions before shuffling. Commit 61837a83 restores that sequence and pins the original field with four exact samples. The focused and full suites are green.
